ostd/src/task/atomic_mode.rs

```diff
@@ -45,3 +45,31 @@ pub fn might_sleep() {
         );
     }
 }
+
+/// A marker trait for guard types that enforce the atomic mode.
+///
+/// Key kernel primitives such as `SpinLock` and `Rcu` rely on
+/// [the atomic mode](crate::task::atomic_mode).
```

Comment:
This does not seem to be a complete sentence. Perhaps say "rely on ... for correctness or soundness".